app/app.py

```diff
@@ -44,19 +58,85 @@ def _get_prediction_endpoint() -> Optional[str]:
     return f"http://{ingress_ip}/seldon/matcha-seldon-workloads/llm/v2/models/transformer/infer"
 
 
-def _create_payload(messages: List[Dict[str, str]]) -> Dict[str, List[Dict[str, str]]]:
+@st.cache_data(show_spinner=False)
+def _get_embedding_function(embed_model_type: str) -> Union[EmbeddingFunction, None]:
+    """Load embedding function to be used by Chroma vector store.
+
+    Args:
+        embed_model_type (str): String representation of the embedding model.
+
+    Returns:
+        Union[EmbeddingFunction, None]: Embedding function if it exists, None otherwise.
+    """
+    # Create a embedding function
+    model_name = EMBED_MODEL_MAP.get(embed_model_type, None)
+    if model_name is None:
+        return None
+    return embedding_functions.InstructorEmbeddingFunction(model_name=model_name)
+
+
+def connect_vector_store(chroma_server_host: str, chroma_server_port: int) -> API:
+    """Connect to Chroma vector store.
+
+    Args:
+        chroma_server_host (str): Chroma server host name
+        chroma_server_port (int): Chroma server port
+
+    Returns:
+        API: Chroma client object.
+    """
+    try:
+        # Connect to vector store
+        chroma_client = ChromaStore(
+            chroma_server_hostname=chroma_server_host,
+            chroma_server_port=chroma_server_port,
+        )
+        return chroma_client
+    except Exception:
+        return None
```

Comment:
Do we need to do something if an exception is thrown?

Reply:
It's raised on line 244